src/eva/core/callbacks/writers/embeddings/classification.py

```diff
@@ -0,0 +1,67 @@
+"""Classification embeddings writer."""
+
+import csv
+import io
+import os
+from typing import Any
+
+import torch
+from torch import multiprocessing
+from typing_extensions import override
+
+from eva.core.callbacks.writers.embeddings import base
+from eva.core.callbacks.writers.embeddings.typings import QUEUE_ITEM
+
+
+class ClassificationEmbeddingsWriter(base.EmbeddingsWriter):
```

Comment:
While the refactoring looks very nice, this seems to undo the changes added in this [PR](https://github.com/kaiko-ai/eva/pull/384), which is needed for WSI level tasks where we need to store all embeddings corresponding to the same slide to a single file?

Reply:
Maybe for that usecase we can implement a `ClassificationMultiEmbeddingsWriter` class or similar?

Reply:
Maybe we can do a `ClassificationMultiEmbeddingsWriter` or similar for that usecase though?

Reply:
hmm I see 🤔 I think `ClassificationMultiEmbeddingsWriter` is good

but also, feel free to commit any modify / add anything that can help with the WSI level or any more refactoring ideas